src/main/java/com/kodilla/ecommercee/controller/ProductController.java

```diff
@@ -1,36 +1,48 @@
 package com.kodilla.ecommercee.controller;
 
 import com.kodilla.ecommercee.dto.ProductDto;
+import com.kodilla.ecommercee.mapper.ProductMapper;
+import com.kodilla.ecommercee.service.ProductService;
+import lombok.RequiredArgsConstructor;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.ArrayList;
 import java.util.List;
 
 @RestController
-@RequestMapping("/v1/products")
+@RequiredArgsConstructor
+@RequestMapping("/v1/product")
 public class ProductController {
 
+    private final ProductMapper productMapper;
+    private final ProductService productService;
+
     @GetMapping
     public List<ProductDto> getProducts() {
-        return new ArrayList<>();
+        return productMapper.mapToProductDtoList(productService.productList());
+    }
+
+    @GetMapping("/{id}")
+    public ProductDto getProduct(@PathVariable Long id) {
+        return productMapper.mapToProductDto(productService.getProductById(id));
     }
 
-    @RequestMapping(method = RequestMethod.GET,  value = "/{productId}")
-    public ProductDto getProduct(@PathVariable Long productId) {
-        return null;
+    @GetMapping("/byname/{name}")
```

Comment:
name to nie id. Lepsza jest ścieżka url typu: "...../byname?name=costam"
Tu lepiej dać @GetMapping("/byname"), a niżej:
getProduct(@RequestParam("name") String name) { ....